OsmAnd/src/net/osmand/plus/routepreparationmenu/RequiredMapsController.java

```diff
@@ -222,4 +218,14 @@ public static void showDialog(@NonNull FragmentActivity activity) {
 		dialogManager.register(PROCESS_ID, new RequiredMapsController(app));
 		RequiredMapsFragment.showInstance(activity.getSupportFragmentManager());
 	}
+
+	public boolean shouldShowOnlineCalculation() {
+		return !onlineCalculationRequested && !isLoadingInProgress() && !isAlreadyCalcOnline();
+	}
+
+	private boolean isAlreadyCalcOnline() {
+		RoutingHelper helper = app.getRoutingHelper();
+		MissingMapsCalculationResult result = helper.getRoute().getMissingMapsCalculationResult();
+		return result != null && result.getMissingMapsPoints().size() > 2 + helper.getIntermediatePoints().size();
```

Comment:
Not a direct sign of calculation type - online or not. 
Perhaps a `String tag` field could be added to `MissingMapsCalculator` and `MissingMapsCalculationResult` to differentiate between types.

Reply:
This is the most obvious method to detect the fact that Calc Online is successfully done already.

We have a number of `missingMapsPoints` and a number of original routing points (`start+end+intermediates = 2 + intermediates`).

Before Calc Online, missingMapsPoints are the same as start+end+intermediates. After, it have a lot of calculated points (hunderds-thousands). Just compare!

The method is based on data that is already exist and, in my opinion, in this particular case we should not expand classes with new useless members just to copy-paste already-existing information.

Reply:
refactores to strange tag